ops/charm.py

```diff
@@ -335,18 +335,15 @@ class CharmBase(Object):
     Args:
         framework: The framework responsible for managing the Model and events for this
             Charm.
-        key: Arbitrary key to distinguish this instance of CharmBase from another.
-            Generally is None when initialized by the framework. For charms instantiated by
-            main.main(), this is currenly None.
     """
 
     on = CharmEvents()
 
-    def __init__(self, framework: Framework, key: typing.Optional[str]):
+    def __init__(self, framework: Framework):
         """Initialize the Charm with its framework and application name.
 
         """
-        super().__init__(framework, key)
+        super().__init__(framework, None)
```

Comment:
I'm happy with this, but it does break everyone who isn't using the *args,**kwargs style of Charm initializers.
Could we do:
```
key=None
```
and then have a deprecation warning if it is being passed?
Or maybe that is
```
**kwargs):

if kwargs:
  DeprecationWarning?
```

Reply:
I will pass some specific object from main, and that will let me understand if we're receiving a parameter because we are passing it ourselves, or because the user is explicitly writing it (in which case we'll issue the warning).